src/optimization/metric_calcs.jl

```diff
@@ -176,8 +176,10 @@ function critical_path(
 
     # Between clusters
     tow_cost = vessel_weightings[1] * mothership_dist_between_clusts(ms_route, num_clusters)
-
-    return cluster_cost_total + tow_cost
+    total_critical_path = cluster_cost_total + tow_cost
+    @assert !isinf(total_critical_path) "Critical path cost is infinite, " *
+                                        "indicating a solution with a bad waypoint"
+    return total_critical_path
```

Comment:
These assertions are great but if its purpose is more than debugging, then I would prefer proper [exception types](https://docs.julialang.org/en/v1/manual/control-flow/#Built-in-Exceptions) be used.

This will allow clearer try/catch patterns to be applied.